packages/rest/src/controllers/didcomm/basic-messages/BasicMessagesController.ts

```diff
@@ -0,0 +1,72 @@
+import { RecordNotFoundError, BasicMessageRole } from '@credo-ts/core'
+import { Body, Controller, Example, Get, Post, Query, Request, Route, Security, Tags } from 'tsoa'
+import { injectable } from 'tsyringe'
+
+import { RequestWithAgent } from '../../../authentication'
+import { apiErrorResponse } from '../../../utils/response'
+import { RecordId, ThreadId } from '../../types'
+
+import { basicMessageRecordExample } from './BasicMessagesControllerExamples'
+import {
+  basicMessageRecordToApiModel,
+  DidCommBasicMessagesSendOptions,
+  type DidCommBasicMessagesRecord,
+} from './BasicMessagesControllerTypes'
+
+@Tags('DIDComm Basic Messages')
+@Route('/didcomm/basic-messages')
+@Security('tenants', ['tenant'])
+@injectable()
+export class DidCommBasicMessagesController extends Controller {
+  /**
+   * Retrieve basic messages by connection id
+   *
+   * @param connectionId Connection identifier
+   * @returns BasicMessageRecord[]
+   */
+  @Example<DidCommBasicMessagesRecord[]>([basicMessageRecordExample])
+  @Get('/')
```

Comment:
Oh thats nice! Good to just use query params as the actual basic message query, hadn't thought about that :).